executor/adapter.go

```diff
@@ -589,6 +589,9 @@ func UpdateForUpdateTS(seCtx sessionctx.Context, newForUpdateTS uint64) error {
 	if !txn.Valid() {
 		return errors.Trace(kv.ErrInvalidTxn)
 	}
+	if seCtx.GetSessionVars().IsIsolation(ast.Serializable) {
```

Comment:
We could add some comments for this and the `We could add some comment` error processing. And we need to think about the documentation for this feature.

Reply:
The documentation is the hard part. It has a variable to disable it by default. Maybe just ignore the documents. Then it will have no effect to all the users.

Reply:
If the user wants to use this feature I think it's better to let them know what and how it works, otherwise they may not know the real effect.

Reply:
PTAL

Reply:
Thanks for your contribution!
We have some high-performance TiDB transaction improvement challenge program, welcome to join the transaction layer optimizations for TiDB.
https://github.com/pingcap/tidb/issues?q=is%3Aopen+is%3Aissue+label%3Ahigh-performance+label%3Asig%2Ftransaction
https://github.com/tikv/tikv/issues?q=is%3Aopen+is%3Aissue+label%3Ahigh-performance+label%3Asig%2Ftransaction.

There are some references in the issue descriptions and a video [link](https://www.bilibili.com/video/BV1vV411m7zK). 
Please let us know if you are interested in these issues.